profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -0,0 +1,186 @@
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact David Poliakoff (dzpolia@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+
+//Modified by Zach Frye at LLNL
+//Contact: frye7@llnl.gov 
+//Organization: CASC at LLNL
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <cstdlib>
+#include <cstring>
+#include <vector>
+#include <unordered_set>
+#include <string>
+#include <regex>
+#include <cxxabi.h>
+#include <dlfcn.h>
+#include <ctime>
+#include <chrono>
+#include <iostream>
+
+//variorum trial run
+extern "C" {
+#include <variorum.h>
+}
+
+bool filterKernels;
+uint64_t nextKernelID;
+std::vector<std::regex> kernelNames;
+std::unordered_set<uint64_t> activeKernels;
+
+typedef void (*initFunction)(const int, const uint64_t, const uint32_t, void*);
+typedef void (*finalizeFunction)();
+typedef void (*beginFunction)(const char*, const uint32_t, uint64_t*);
+typedef void (*endFunction)(uint64_t);
+
+static initFunction initProfileLibrary = NULL;
+static finalizeFunction finalizeProfileLibrary = NULL;
+static beginFunction beginForCallee = NULL;
+static beginFunction beginScanCallee = NULL;
+static beginFunction beginReduceCallee = NULL;
+static endFunction endForCallee = NULL;
+static endFunction endScanCallee = NULL;
+static endFunction endReduceCallee = NULL;
+
+//variables for simple timer
+time_t start_time;
+
+
+extern "C" void kokkosp_init_library(const int loadSeq,
+	const uint64_t interfaceVer, const uint32_t devInfoCount, void* deviceInfo) {
+    
+    int ret;
+    //Simple timer code to keep track of the general amount of time the application ran for.
+    time(&start_time);
+    std::cout << "Start Time: " << start_time << std::endl;
```

Comment:
I wouldn't flush the stream here.
```suggestion
    std::cout << "Start Time: " << start_time << '\n';
```
or use `printf`